### PR TITLE
feat: added example and explanation for using slots in compound variants

### PR DESCRIPTION
**Pull request by gijsbotje**
Added a section to the slots page to show how to use slots with compound variants.
image

Feel free to take over and change the example for a different one.

### CODE REVIEW

File: pages/docs/slots.mdx [hunk 1]
```diff
@@ -324,6 +325,188 @@ export default App;
...
+      }
+    }
+  ],
+  defaultVariants: {
+    variant: 'filled',
+    severity: 'success'
+  }
+});
+
+const Alert = ({ severity, variant }) => {
```
Comment (mskelton) [inline pages/docs/slots.mdx:405]: Let's remove the component and just keep the example simple like the `Slots with responsive variants` example

File: pages/docs/slots.mdx [hunk 2]
```diff
@@ -324,6 +325,188 @@ export default App;
   </Tab>
 </Tabs>
 
+### Slots with compound variants
+
+Like variants, you can also define the styles for each slot when using compound variants. The slots can be passed to the `class` as an object.
```
Comment (mskelton) [inline pages/docs/slots.mdx:330]:
```suggestion
Like variants, you can also define the styles for each slot when using compound variants. The slots can be passed to the `class` or `className` as an object.
```

File: components/examples/slots-compound-variants-example.tsx [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+    },
+    {
+      variant: 'filled',
+      severity: 'success',
+      class: {
+        root: 'bg-green-500',
+        title: 'text-green-100',
+        message: 'text-green-200'
+      }
+    }
```
Comment (mskelton) [inline components/examples/slots-compound-variants-example.tsx:64]: The color contrast of the title and message for the filled variant is very low.  We should use colors with more contrast to make the text accessible for all users.

image

image

**Comment by gijsbotje** [comment]
@mskelton Thanks for the feedback! I just pushed the requested changes, let me know if I need to change more.

**Comment by mskelton** [comment]
Thanks for the contribution!

**Review by mskelton — APPROVED** [review]

**Pull request merged**